lib/unexpected-immutable.js

```diff
@@ -24,6 +24,16 @@ module.exports = {
     expect.addAssertion('<Immutable> to [exhaustively] satisfy <any>', function (expect, subject, pattern) {
       return expect(subject.toJS(), 'to [exhaustively] satisfy', pattern)
     })
+    expect.addAssertion('<Immutable> to have deep value <array> <any?>', function (expect, subject, path, value) {
```

Comment:
Hmm... Thinking of unexpected's capabilities, it'd probably work better to have an intermediate assertion like `<Immutable> when following <array>', which then gets the contents of that location and shifts it to following assertions? I'll look at it next week.